src/Compilers/Server/VBCSCompiler/MetadataCache.cs

```diff
@@ -74,10 +72,11 @@ internal Metadata GetMetadata(string fullPath, MetadataReferenceProperties prope
             }
             else
             {
+                Debug.Assert(fileKey.HasValue);
                 var primaryModule = CreateModuleMetadata(fullPath, prefetchEntireImage: false);
 
                 // Get all the modules, and load them. Create an assembly metadata.
-                var allModules = GetAllModules(primaryModule, Path.GetDirectoryName(fullPath));
+                var allModules = GetAllModules(primaryModule, Path.GetDirectoryName(fullPath)!);
```

Comment:
>Path.GetDirectoryName(fullPath)! [](start = 62, length = 32)

Why does this need a suppression?

Reply:
Because `GetDirectoryName` returns `string?` (both now and in net5.0).